lib/pbench/server/filetree.py

```diff
@@ -257,7 +257,7 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
     def extract(self, path: str) -> str:
         """
-        Extract a file from the tarball and return it as a string
```

Comment:
Wouldn't we want to document this as "return it's contents as a string."?

And are we sure we want to alway decode the contents?

Reply:
Adding "contents" is probably good; adding the apostrophe is not.  🙂

As for decoding, that's an interesting question.  I'm going to defer to Dave on this one, as I started digging into this and the hole got too deep.  Depending on what `extractfile()` returns, `read()` is likely to return an "encoded" string which is possibly _not_ what we want to return to the requestor, and therefore we probably should always decode it.  However, if we were to specify the encoding (or, perhaps, read the file in binary mode?), then we wouldn't need the decode step because it wouldn't have been encoded....

Reply:
This was not intended to be a full general artifact access method. It's also not new code, and I only brought attention to it because for some bizarre reason I tweaked the docstring. I don't intend to generalize it further in this context.

It exists so that the `PUT` code could extract `metadata.log` before we unpacked the tarball. I expect this will become a call into the cache manager, which will definitely need to handle binary files (e.g., the tarball itself) as well as text files. We may or may not need to support different encodings, or specify `utf-8` and be done with it (if we know everything we care about is ASCII or Unicode).

None of this, luckily, is relevant to this PR except wordsmithing the docstring... and I'm beginning to think I should just revert the change and pretend I never applied the can opener to this particular can of worms ...